src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash128.cs

```diff
@@ -0,0 +1,460 @@
+                ulong* accumulators = stackalloc ulong[AccumulatorCount];
+                CopyAccumulators(ref _state, accumulators);
+
+                fixed (byte* secret = _state.Secret)
+                {
+                    DigestLong(ref _state, accumulators, secret);
+                    current.Low64 = MergeAccumulators(accumulators, secret + SecretMergeAccsStartBytes, _state.TotalLength * Prime64_1);
+                    current.High64 = MergeAccumulators(accumulators, secret + SecretLengthBytes - AccumulatorCount * sizeof(ulong) - SecretMergeAccsStartBytes, ~(_state.TotalLength * Prime64_2));
+                }
+            }
+            else
+            {
+                fixed (byte* buffer = _state.Buffer)
+                {
+                    current = HashToHash128(new ReadOnlySpan<byte>(buffer, (int)_state.TotalLength), (long)_state.Seed);
+                }
+            }
+
+            return current;
+        }
+
+#if NET7_0_OR_GREATER
+        /// <summary>Gets the current computed hash value without modifying accumulated state.</summary>
+        /// <returns>The hash value for the data already provided.</returns>
+        [CLSCompliant(false)]
+        public UInt128 GetCurrentHashAsUInt128()
+        {
+            Hash128 current = GetCurrentHashAsHash128();
+            return new UInt128(current.High64, current.Low64);
+        }
+#endif
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void WriteBigEndian128(in Hash128 hash, Span<byte> destination)
+        {
+            // TODO https://github.com/dotnet/runtime/issues/72107: Use BinaryPrimitives.WriteUInt128BigEndian() once it is supported
+            ulong low = hash.Low64;
+            ulong high = hash.High64;
+            if (BitConverter.IsLittleEndian)
+            {
+                low = BinaryPrimitives.ReverseEndianness(low);
+                high = BinaryPrimitives.ReverseEndianness(high);
+            }
+
+            ref byte dest0 = ref MemoryMarshal.GetReference(destination);
+            Unsafe.WriteUnaligned(ref dest0, high);
+            Unsafe.WriteUnaligned(ref Unsafe.AddByteOffset(ref dest0, new IntPtr(sizeof(ulong))), low);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Hash128 HashLength0To16(byte* source, uint length, ulong seed)
+        {
+            if (length > 8)
+            {
+                return HashLength9To16(source, length, seed);
+            }
+
+            if (length >= 4)
+            {
+                return HashLength4To8(source, length, seed);
+            }
+
+            if (length != 0)
+            {
+                return HashLength1To3(source, length, seed);
+            }
+
+            const ulong bitflipl = DefaultSecretUInt64_8 ^ DefaultSecretUInt64_9;
+            const ulong bitfliph = DefaultSecretUInt64_10 ^ DefaultSecretUInt64_11;
```

Comment:
Nit: we use PascalCasing for consts, even locals

Reply:
Fixed by commit 4505e57